homeassistant/components/baf/fan.py

```diff
@@ -0,0 +1,101 @@
+"""Support for Big Ass Fans fan."""
+from __future__ import annotations
+
+import math
+from typing import Any
+
+from aiobafi6 import Device, OffOnAuto
+
+from homeassistant import config_entries
+from homeassistant.components.fan import (
+    DIRECTION_FORWARD,
+    DIRECTION_REVERSE,
+    FanEntity,
+    FanEntityFeature,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util.percentage import (
+    percentage_to_ranged_value,
+    ranged_value_to_percentage,
+)
+
+from .const import DOMAIN, PRESET_MODE_AUTO, SPEED_COUNT, SPEED_RANGE
+from .entity import BAFEntity
+from .models import BAFData
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: config_entries.ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up SenseME fans."""
+    data: BAFData = hass.data[DOMAIN][entry.entry_id]
+    if data.device.has_fan:
+        async_add_entities([BAFFan(data.device)])
+
+
+class BAFFan(BAFEntity, FanEntity):
+    """BAF ceiling fan component."""
+
+    _attr_supported_features = FanEntityFeature.SET_SPEED | FanEntityFeature.DIRECTION
+    _attr_preset_modes = [PRESET_MODE_AUTO]
+
+    def __init__(self, device: Device) -> None:
+        """Initialize the entity."""
+        super().__init__(device, device.name)
+        self._attr_speed_count = SPEED_COUNT
```

Comment:
Why not assign this one above like the others?

Reply:
copypasta. Will fix and drop the useless init

Reply:
65c17988785c6220567f87a668e6fc5b7660b70b